views/pages/AdminPage/index.jsx

```diff
@@ -34,15 +34,24 @@ const AdminPage = () => {
   const [currentRowIndex, setCurrentRowIndex] = useState(null)
   const [openAccessModal, setOpenAccessModal] = useState(false)
 
-  const filteredUsers = (searchOptionsValue || []).length ?
-    users.filter(user => searchOptionsValue.map(searchOption => searchOption.value).includes(user.uid)) :
-    users
+  const filteredUsers = (searchOptionsValue || []).length
+    ? users.filter((user) =>
+        searchOptionsValue
+          .map((searchOption) => searchOption.value)
+          .includes(user.uid)
+      )
+    : users
 
   const updateUsers = (newUser) => {
-    const newUserIndex = users.findIndex(u => u.uid === newUser.uid) 
+    const newUserIndex = users.findIndex((u) => u.uid === newUser.uid)
     if (newUserIndex === -1) return
 
-    setUsers(users.slice(0, newUserIndex).concat([newUser]).concat(users.slice(newUserIndex + 1)))
+    setUsers(
+      users
```

Comment:
Can we put this operation into a descriptive helper method like `insertUser`?

Reply:
The parent function updatedUsers gets reused in the code, I've moved the setUsers logic into a constant for readability. 

Is the way that the updatedUsers function being used is in the ballpark of what you're suggesting..?

Reply:
[7134be9](https://github.com/AMP-SCZ/dpdash/pull/564/commits/7134be95aa654666d73a2e5827295f698d8b9acb)

Reply:
Yeah that looks great!